frontend/components/forms/fields/Checkbox/Checkbox.tsx

```diff
@@ -36,22 +37,41 @@ const Checkbox = (props: ICheckboxProps) => {
     name,
     onChange = noop,
     onBlur = noop,
-    value,
+    value = false,
     wrapperClassName,
-    indeterminate,
+    indeterminate = false,
     parseTarget,
     tooltipContent,
     isLeftLabel,
     helpText,
   } = props;
 
-  const handleChange = () => {
+  const handleChange = (
+    event: React.MouseEvent | React.KeyboardEvent
+  ): void => {
+    event.preventDefault();
+    if (readOnly || disabled) return;
+
+    let newValue: boolean | null;
+    if (indeterminate) {
+      newValue = true;
+    } else if (value === true) {
+      newValue = false;
+    } else {
+      newValue = true;
+    }
+
     if (parseTarget) {
-      // Returns both name and value
-      return onChange({ name, value: !value });
+      onChange({ name, value: newValue });
+    } else {
+      onChange(newValue);
     }
+  };
 
-    return onChange(!value);
+  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>): void => {
```

Comment:
potentially remove this as onEnter might be assigned to submit buttons, then figure out a way to only have onEnter on tables